controller/matchController.js

```diff
@@ -0,0 +1,64 @@
+var Parse = require('parse/node').Parse;
+var parseAppId = process.env.PARSE_APP_ID;
+Parse.initialize(parseAppId);
+Parse.serverURL = process.env.SERVER_URL;
+
+const matchController = {
+  findMatches: async function (event, seeker, limit = 3, radius_km) {
+
+    if (event == undefined) {
+      throw ({ code: 400, message: "No event provided" });
+    }
+
+    if (seeker == undefined) {
+      throw ({ code: 400, message: "No seeker provided" });
+    }
+
+    const event_type = event.get("event_type");
+    if (event_type == undefined) {
+      throw ({ code: 400, message: "No event type provided" });
+    }
+
+    const seeker_location = seeker.get("location");
+    if (event_type == undefined) {
+      throw ({ code: 400, message: "No event type provided" });
+    }
+
+    // Optional
+    const special_needs = seeker.get("special_needs");
+
+    // Construct inner query
+    const Volunteer = Parse.Object.extend("Volunteer");
+    const innerQuery = new Parse.Query(Volunteer);
```

Comment:
Here's we're making a query on the Volunteer class, taking into consideration the event's `event_type` and the seeker's `special_needs`. We're also restricting the results to only show screened volunteers.